raiden/transfer/mediated_transfer/mediation_fee.py

```diff
@@ -79,11 +79,13 @@ def _collect_x_values(
 
     To find the penalty fee for the current transfer.
     """
-    all_x_vals = [x - balance_in for x in penalty_func_in.x_list] + [
+    all_x_vals: List[Fraction] = [x - balance_in for x in penalty_func_in.x_list] + [
         balance_out - x for x in penalty_func_out.x_list
     ]
-    limited_x_vals = (max(min(x, balance_out, max_x), 0) for x in all_x_vals)
-    return sorted(set(Fraction(x) for x in limited_x_vals))
+    limited_x_vals = (
+        max(min(x, Fraction(balance_out), Fraction(max_x)), Fraction(0)) for x in all_x_vals
```

Comment:
`min` and `max` between `int`s and `Fraction`s should be a valid, safe operation. `Fraction` is not just a mypy type cast, but a real python constructor, so I'm afraid this might slow down these operations. This is called repeatedly during the mediation fee calculation, so performance can be relevant here. If mypy mistakenly complains about a `max(int, Fraction)`, consider using a type ignore.

Reply:
Mypy was happy, I didn't think about performance implications.
But you're right, it's a type-stub issue, and type conversion can be bad here.